src/ArcherElement.js

```diff
@@ -3,10 +3,13 @@ import PropTypes from 'prop-types';
 
 export class ArcherElement extends React.Component {
   componentWillReceiveProps(nextProps) {
-    if (
-      JSON.stringify(this.props.relations) ===
-      JSON.stringify(nextProps.relations)
-    ) {
+    const currentRelations = (this.props.relations || []).map(r =>
+      (r.label || {}).toString(),
+    );
+    const nextRelations = (nextProps.relations || []).map(r =>
+      (r.label || {}).toString(),
+    );
```

Comment:
We could do a `stringifyRelation` function called twice to make this clearer 😊

Reply:
@jrfferreira Ouch, it turns out this function was wrong. We lose the 'to' and 'from' in the stringification. Besides, `toString()` always returns `[Object object]` 😉 
I'll fix this (already merged unfortunately)

Reply:
Oh, sorry about that! I saw the fix too, thanks!